### PR TITLE
[APB-3624][MP] Add endpoint to get agencyEmail from DES

### PR DESCRIPTION
**Pull request by mariannepearson0**

### CODE REVIEW

File: app/uk/gov/hmrc/agentservicesaccount/connectors/DesConnector.scala [hunk 1]
```diff
@@ -93,27 +93,34 @@ class DesConnector @Inject()(
 
   override val kenshooRegistry: MetricRegistry = metrics.defaultRegistry
 
-  def getAgencyName(
-    clientIdentifier: TaxIdentifier)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[String]] = {
-
-    val url = clientIdentifier match {
+  private def getAgentRecordUrl(clientId: TaxIdentifier) =
```
Comment (johnrevill) [inline app/uk/gov/hmrc/agentservicesaccount/connectors/DesConnector.scala]: pedantic: The `clientIdentifier` seems like a misnomer here as it's an identifier used to retrieve agency details. Same on lines 109 and 119.

File: conf/app.routes [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # microservice specific routes
 
 GET         /agent/agency-name                           @uk.gov.hmrc.agentservicesaccount.controllers.AgentServicesAccountController.getCurrentAgencyName
+GET         /agent/agency-email                          @uk.gov.hmrc.agentservicesaccount.controllers.AgentServicesAccountController.getCurrentAgencyEmail
```
Comment (johnrevill) [inline conf/app.routes]: Endpoint behaviour needs documenting in README.md

File: it/uk/gov/hmrc/agentservicesaccount/connectors/DesConnectorISpec.scala [hunk 3]
```diff
@@ -106,23 +106,26 @@ class DesConnectorISpec
     behave like testGetAgencyNameMethod(utr)
 
     def testGetAgencyNameMethod(identifier: TaxIdentifier) = {
```
Comment (johnrevill) [inline it/uk/gov/hmrc/agentservicesaccount/connectors/DesConnectorISpec.scala]: pedantic: update method name to reflect it tests agency email too?

File: it/uk/gov/hmrc/agentservicesaccount/controllers/AgentServicesAccountControllerISpec.scala [hunk 4]
```diff
@@ -189,6 +195,34 @@ class AgentServicesAccountControllerISpec extends ServerBaseISpec with DataStrea
     }
   }
 
+  "GET /agent/agency-email" should {
+    "return Ok and agencyName after successful response" in {
```
Comment (johnrevill) [inline it/uk/gov/hmrc/agentservicesaccount/controllers/AgentServicesAccountControllerISpec.scala]: pedantic: "agencyName" -> "agencyEmail"

File: conf/app.routes (hunk 2 above)
Comment (mariannepearson0) [reply conf/app.routes]: 👍

File: it/uk/gov/hmrc/agentservicesaccount/connectors/DesConnectorISpec.scala (hunk 3 above)
Comment (mariannepearson0) [reply it/uk/gov/hmrc/agentservicesaccount/connectors/DesConnectorISpec.scala]: 👍

File: it/uk/gov/hmrc/agentservicesaccount/controllers/AgentServicesAccountControllerISpec.scala (hunk 4 above)
Comment (mariannepearson0) [reply it/uk/gov/hmrc/agentservicesaccount/controllers/AgentServicesAccountControllerISpec.scala]: 👍

File: app/uk/gov/hmrc/agentservicesaccount/connectors/DesConnector.scala (hunk 1 above)
Comment (mariannepearson0) [reply app/uk/gov/hmrc/agentservicesaccount/connectors/DesConnector.scala]: 👍

**Pull request merged**